services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py

```diff
@@ -351,20 +351,35 @@ async def remove_orphaned_services(
         node_ids = await get_workbench_node_ids_from_project_uuid(app, project_uuid)
         currently_opened_projects_node_ids.update(node_ids)
 
-    running_interactive_services = await get_running_interactive_services(app)
+    running_interactive_services: List[
+        Dict[str, Any]
+    ] = await get_running_interactive_services(app)
     logger.info(
         "Will collect the following: %s",
         [x["service_host"] for x in running_interactive_services],
     )
     for interactive_service in running_interactive_services:
         # if not present in DB or not part of currently opened projects, can be removed
         node_id = interactive_service["service_uuid"]
-        if (
-            not await is_node_id_present_in_any_project_workbench(app, node_id)
-            or node_id not in currently_opened_projects_node_ids
-        ):
+        # if the node does not exist in any project in the db, we can safely remove it without saving any state
+        if not await is_node_id_present_in_any_project_workbench(app, node_id):
```

Comment:
This will fix the case we found the other day with TN. He opens a jupytelab node, deletes it and creates a new one. The deleted node remained in the swarm forever

Reply:
nice one!